pkg/op/storage.go

```diff
@@ -16,12 +16,22 @@ type AuthStorage interface {
 	SaveAuthCode(context.Context, string, string) error
 	DeleteAuthRequest(context.Context, string) error
 
-	CreateAccessToken(context.Context, TokenRequest) (string, time.Time, error)
-	CreateAccessAndRefreshTokens(ctx context.Context, request TokenRequest, currentRefreshToken string) (accessTokenID string, newRefreshToken string, expiration time.Time, err error)
-	TokenRequestByRefreshToken(ctx context.Context, refreshToken string) (RefreshTokenRequest, error)
+	// The TokenRequest parameter of CreateAccessToken can be any of:
+	// - TokenRequest as returned by ClientCredentialsStorage.ClientCredentialsTokenRequest
+	// - RefreshTokenRequest as returned by AuthStorage.TokenRequestByRefreshToken
+	// - AuthRequest as returned one of the AuthStorage methods above
+	// - *oidc.JWTTokenRequest created by decoding a JWT
```

Comment:
the JWT is the `assertion` value of a JWT Profile Grant: https://datatracker.ietf.org/doc/html/rfc7523#section-2.1

Reply:
noted